src/sentry/utils/arroyo.py

```diff
@@ -51,3 +62,105 @@ def timing(self, name: str, value: Union[int, float], tags: Optional[Tags] = Non
         self.__backend.timing(
             key=self.__merge_name(name), value=value, tags=self.__merge_tags(tags)
         )
+
+
+def _get_arroyo_subprocess_initializer(
+    initializer: Optional[Callable[[], None]]
+) -> Callable[[], None]:
+    from sentry.metrics.middleware import get_current_global_tags
+
+    # One can add integer tags and other invalid types today. Filter out any
+    # tags that may not be pickleable. Because those tags are getting pickled
+    # as part of the constructed partial()
+    tags: Tags = {k: v for k, v in get_current_global_tags().items() if isinstance(v, str)}
+    return partial(_initialize_arroyo_subprocess, initializer=initializer, tags=tags)
+
+
+def _initialize_arroyo_subprocess(initializer: Optional[Callable[[], None]], tags: Tags) -> None:
+    from sentry.runner import configure
+
+    configure()
+
+    if initializer:
+        initializer()
+
+    from sentry.metrics.middleware import add_global_tags
+
+    # Inherit global tags from the parent process
+    add_global_tags(_all_threads=True, **tags)
+
+    _initialize_arroyo_main()
+
+
+def _initialize_arroyo_main() -> None:
+    from arroyo import configure_metrics
+
+    from sentry.utils.metrics import backend
+
+    metrics_wrapper = MetricsWrapper(backend, name="consumer")
+    configure_metrics(metrics_wrapper)
```

Comment:
Should we skip this in the subprocess initialization? Since Arroyo never fires metrics from there it doesn't do anything.